http/src/main/java/io/micronaut/http/uri/UriMatchTemplate.java

```diff
@@ -61,39 +55,41 @@ protected UriMatchTemplate(CharSequence templateString, Object... parserArgument
         super(templateString, parserArguments);
 
         this.matchPattern = Pattern.compile(pattern.toString());
-        this.variables = variableList.toArray(new String[variableList.size()]);
+        this.variables = variablesModifier.keySet().toArray(new String[0]);
         String tmpl = templateString.toString();
         int len = tmpl.length();
         this.isRoot = len == 0 || (len == 1 && tmpl.charAt(0) == '/');
         // cleanup / reduce memory consumption
         this.pattern = null;
-        this.variableList = null;
     }
 
     /**
-     * @param templateString The template
-     * @param segments       The list of segments
-     * @param matchPattern   The match pattern
-     * @param variables      The variables
+     * @param templateString    The template
+     * @param segments          The list of segments
+     * @param matchPattern      The match pattern
+     * @param variablesModifier The modifier char for each variable in route
+     * @param variables         The variables
      */
-    protected UriMatchTemplate(CharSequence templateString, List<PathSegment> segments, Pattern matchPattern, String... variables) {
+    protected UriMatchTemplate(CharSequence templateString, List<PathSegment> segments, Pattern matchPattern, Map<String, Character> variablesModifier, String... variables) {
```

Comment:
Would the variables be duplicated here? Why pass in both objects?

Reply:
Since, `variables` and `variableModifiers` are two different properties inside UriMatchTemplate. And by looking [here](https://github.com/micronaut-projects/micronaut-core/blob/master/http/src/main/java/io/micronaut/http/uri/UriMatchTemplate.java#L171) I was unsure if it's a good idea to remove variables from the constructor.